### ISSUE TITLE
I believe it collides with KASHacters Multi Character

### ISSUE THREAD

**Issue by MoravianLion** [Xseba360/esx_locksystem#11]
I think it doesn't work anymore with multi character support.

It just says "Keys not found" and then "Vehicle isn't yours"

https://forum.fivem.net/t/release-esx-kashacters-multi-character/251613

Please, pretty please, try to include support for this in next update!

**Comment by Pesoen** [comment]
i agree, this happens on our server as well. i am working on trying to fix it.. i think the problem is it loads the character identifier as soon as you join, instead of when the player is spawned. if i manage to make it. i will create a pull request.

will make a setting in config you can enable to make it supported. no promises on when its done though..

**Comment by Agony099** [comment]
> i agree, this happens on our server as well. i am working on trying to fix it.. i think the problem is it loads the character identifier as soon as you join, instead of when the player is spawned. if i manage to make it. i will create a pull request.
> 
> will make a setting in config you can enable to make it supported. no promises on when its done though..

That would be amazing

**Comment by Pesoen** [comment]
> > i agree, this happens on our server as well. i am working on trying to fix it.. i think the problem is it loads the character identifier as soon as you join, instead of when the player is spawned. if i manage to make it. i will create a pull request.
> > will make a setting in config you can enable to make it supported. no promises on when its done though..
> 
> That would be amazing

so far its not working.. i have tried making modifications to just this part of the entire thing, but i might need to make some modifications on the kashacters side of things, so it calls the function from there instead.. gonna experiment with this a bit longer though, would be best to just make the changes to one part of it, rather than having to make it on two separate scripts.

However i have managed to get a ton of debugging information to display, which helps me see when it loads the keys, and what keys its loading. so it will not take super long to make the required changes once i can figure out what part to hook it into, or if i need to add a line of code to the kashacters script.

Either way, i am working on it, but exams are taking a ton of my time currently.

**Comment by KoolAiddTV** [comment]
has anything fixed this?

**Comment by iSentrie** [comment]
Never had such problem, lol. And u just have to put some haskeys function on owned vehicles.

**Comment by aboodilatif** [comment]
Replaced 
`AddEventHandler("playerSpawned", function()`
line 18 in client.lua with:
`AddEventHandler("es:allowedToSpawn", function()`

not sure if it will work with everyone
